File: renders/facilities/Restaurant.cpp
/* NIM/Nama  : 13515AAA / AAA AAA   */
/* File      : Restaurant.cpp       */

#include <iostream>
#include "Restaurant.h"
using namespace std;

/***** RESTAURANT *****/
Restaurant::Restaurant() : Facility('R', WHITE)
{}

Restaurant::Restaurant(const Restaurant& R) : Facility(R)
{}

Restaurant::~Restaurant()
{}

Restaurant& Restaurant::operator=(const Restaurant& R)
{
  Facility::operator=(R);
  return *this;
}

Restaurant* Restaurant::clone() const {
  return new Restaurant(*this);
}

File: Restaurant.java
/**
 * @author NIM / Nama : 13515087 / Audry Nyonata
 *  Nama file : Restaurant.java
 */

import java.util.*;
import java.lang.*;

public class Restaurant extends Facility{
   /**
   * Constructor
   */
  public Restaurant(){
    super();
  }
  /**
   * Mengembalikan karakter untuk ditampilkan ke layar
   * @return id bertipe char
   */
  public char render(){
    return 'R';
  }
  /**
   * Mengembalikan warna untuk ditampilkan ke layar
   * @return color warna renderable
   */
  public String renderWithColor(){
    return ANSI_WHITE + render() + ANSI_RESET;
  }
}
